packages/core/test/utils_test.js

```diff
@@ -3714,4 +3715,173 @@ describe("utils", () => {
       expect(schemaRequiresTrueValue({ type: "string" })).eql(false);
     });
   });
+
+  describe("getMatchingOption", () => {
+    describe("basic example", () => {
+      const options = Object.freeze([
+        {
+          type: "object",
+          properties: {
+            type: {
+              type: "string",
+              enum: ["One"],
+              default: "One",
+            },
+          },
+          required: ["type"],
+        },
+        {
+          type: "object",
+          properties: {
+            type: {
+              type: "string",
+              enum: ["Two"],
+              default: "Two",
+            },
+          },
+          required: ["type"],
+        },
+      ]);
+
+      it("returns undefined (default option) if nothing matches the schema", () => {
+        const data = { type: "Three" };
+
+        expect(getMatchingOption(data, options)).eql(undefined);
+      });
+
+      it("returns the expected option if it matches the schema", () => {
+        const data = { type: "Two" };
+
+        expect(getMatchingOption(data, options)).eql(1);
+      });
+    });
+
+    describe("partial form data", () => {
+      it("returns the expected option for a partial match", () => {
+        const options = Object.freeze([
```

Comment:
I used `Object.freeze` because we need to make sure we don't mutate the original schema by accident

Reply:
(it throws an exception if you mutate the object)

<img width="703" alt="Screenshot 2020-12-02 at 15 15 18" src="https://user-images.githubusercontent.com/17656/100891511-3cde7600-34b1-11eb-8d2d-b9ae1503631a.png">